titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/controller/NodeGcController.java

```diff
@@ -148,9 +150,16 @@ private boolean isAgentInstanceNotAvailable(String nodeName) {
 
         Optional<AgentInstance> agentInstanceOpt = agentManagementService.findAgentInstance(nodeName);
         if (!agentInstanceOpt.isPresent()) {
-            return true;
+            // re-check with agent service to find an instance
+            AgentInstance agentInstance = agentManagementService.getAgentInstanceAsync(nodeName)
+                    .toBlocking()
+                    .firstOrDefault(null);
+            if (agentInstance != null && agentInstance.getLifecycleStatus() != null) {
```

Comment:
If after direct query agentInstance is null we should assume that it is not available, so check it first and return `true`?

Reply:
yes, this could should also handle the case of the instance is explicitly gone in AWS. This assumes you can query for the state forever but the EC2 API returns a 404 after the instance has been stopped for an hour IIRC.